mlflow/store/jobs/abstract_store.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+from typing import Iterator
+
+from mlflow.entities._job import Job
+from mlflow.entities._job_status import JobStatus
+from mlflow.utils.annotations import developer_stable
+
+
+@developer_stable
+class AbstractJobStore(ABC):
+    """
+    Abstract class that defines API interfaces for storing Job metadata.
+    """
+
+    @abstractmethod
+    def create_job(self, function_fullname: str, params: str, timeout: float | None = None) -> Job:
+        """
+        Create a new job with the specified function and parameters.
+
+        Args:
+            function_fullname: The full name of the function to execute
+            params: The job parameters that are serialized as a JSON string
+            timeout: The job execution timeout in seconds
+
+        Returns:
+            Job entity instance
+        """
+
+    @abstractmethod
+    def start_job(self, job_id: str) -> None:
+        """
+        Start a job by setting its status to RUNNING.
+
+        Args:
+            job_id: The ID of the job to start
+        """
+
+    @abstractmethod
+    def reset_job(self, job_id: str) -> None:
```

Comment:
```suggestion
    def set_job_status(self, job_id: str) -> None:
```

we can reuse this method in the abstract class to set the job status.